### ISSUE TITLE
Issue with markup

### ISSUE THREAD

**Issue by SNTagore** [joaotavora/yasnippet#999]
After enabling markup in markdown-mode, yasnippets has responded weirdly to tab jump points (meaning $1, $2, $0, etc).  

For instance, when using yasnippet `link` with code `[${1:Link Text}](${2:URL} $3) $0`, cursor will start at $1.  Pressing tab with markup enabled with cause a jump to the end of the snippet (not $0, but after it, confirmed by playing with the code on my system).

**Comment by npostavs** [comment]
Right, I think we need to disable invisibility within snippets, see #1001.
